apdbPlots.py

```diff
@@ -0,0 +1,599 @@
+    srcRas = sourceTable.loc[sourceTable['diaObjectId'] == objId, 'ra'].values
+    srcDecs = sourceTable.loc[sourceTable['diaObjectId'] == objId, 'decl'].values
+    srcCcdVisitIds = sourceTable.loc[sourceTable['diaObjectId'] == objId, 'ccdVisitId'].values
+    dataIds = [{'visit': int(str(dataId)[0:6]), 'ccdnum': int(str(dataId)[6:])} for dataId in srcCcdVisitIds]
+
+    objLoc = lsst.geom.SpherePoint(objRa, objDec, lsst.geom.degrees)
+    calexpFirst = butler.get('calexp', dataId=dataIds[0])
+    templateCutout = plc.getTemplateCutout(calexpFirst, repo, objLoc)
+    wcs = templateCutout.getWcs()
+
+    disp = afwDisplay.Display(plotIdx, reopenPlot=True)
+    disp.setMaskTransparency(100)
+    disp.scale('asinh', 'zscale', Q=4)
+    disp.mtv(templateCutout, title=objId)
+
+    with disp.Buffering():  # obviously
+        coordObj = wcs.skyToPixel(objLoc)
+        disp.dot('+', *coordObj, ctype='C0', size=5)
+        for dataId, srcId, ra, dec in zip(dataIds, srcIds, srcRas, srcDecs):
+            calexp = butler.get('calexp', dataId=dataId)
+            psf = calexp.getPsf()
+            psfSize = psf.computeShape().getDeterminantRadius()
+            coordSrc = wcs.skyToPixel(lsst.geom.SpherePoint(ra, dec, lsst.geom.degrees))
+            if goodSrc is not None:
+                if srcId in goodSrc['diaSourceId'].values:
+                    disp.dot('o', *coordSrc, ctype='C3', size=psfSize)  # green
+                else:
+                    disp.dot('o', *coordSrc, ctype='C2', size=psfSize)  # red
+            else:
+                disp.dot('o', *coordSrc, ctype='C0', size=psfSize)  # blue
+
+
+def plotSeeingHistogram(repo, sourceTable, ccd=35, ccdXcenter=1000,
+                        ccdYcenter=2000, pixelScale=0.26, title=''):
+    """Plot distribution of visit seeing.
+
+    Parameters
+    ----------
+    repo : `str`
+        Repository corresponding to the output of an ap_pipe run.
+    sourceTable : `pandas.core.frame.DataFrame`
+        Pandas dataframe with DIA Sources from an APDB.
+    ccd : `int`
+        The ccd being considered.
+    ccdXcenter : `int`
+        Pixel center of ccd in x-direction (e.g., 1000 for 2000x4000 ccd).
+    ccdYcenter : `int`
+        Pixel center of ccd in y-direction (e.g., 2000 for 2000x4000 ccd).
+    pixelScale : `float`
+        Pixel scale of the instrument in arcseconds per pixel, default 0.26.
+    title : `str`
+        Title for the plot, optional.
+    """
+    fwhm = pd.DataFrame()
+    sourceTable = addVisitCcdToSrcTable(sourceTable)
+    butler = dafPersist.Butler(repo)
+    visits = np.unique(sourceTable['visit'])
+    radii = []
+    for visit in visits:
+        miniCalexp = butler.get('calexp_sub', visit=int(visit),
+                                ccd=ccd, bbox=lsst.geom.Box2I())
+        psf = miniCalexp.getPsf()
+        shape = psf.computeShape(lsst.geom.Point2D(ccdXcenter, ccdYcenter))
```

Comment:
Why is this psf position dependent when the one above isn't? How come this one uses a 'calexp_sub' and the other one doesn't? Can this be done from a catalogue?

Reply:
Because this one originated from Yusra and the other one didn't 😁 I think this is an equivalent way to get the PSF for each CCD. Maybe it's faster since it only loads a calexp_sub instead of a full calexp? I am pretty darn sure PSFs are per-CCD, and you can apparently retrieve one either by throwing a coordinate at it or by asking it to retrieve it for a calexp directly. Yusra was thinking in terms of sky coordinates and I was thinking in terms of exposures. Although, entirely unrelatedly, now that I look at this more closely, it might be good to use the FWHM above instead of the radius, so I'll at least do that!